### DISCUSSION TITLE
Sonarr 4

### DISCUSSION THREAD

**Discussion by Dulanic** [tteck/Proxmox#1738] [Ideas]
Have you considered adding sonarr 4 as an option for sonarr install? It is so much better, and I was just transitioning to proxmox from docker, and I used it there. I only realized this wasn't the same version when I restored my backup, and it complained about a missing column :) Maybe making it a choice as to which version to use during install?

**Comment by tteck** [comment]
Is there an install path? All that I can find is
```
wget -q https://download.sonarr.tv/v4/develop/4.0.0.646/Sonarr.develop.4.0.0.646.linux-x64.tar.gz
tar -xzf Sonarr.develop.4.0.0.646.linux-x64.tar.gz
```
Which looks like you need to move/overwrite files.

**Reply by DocDrydenn** [reply]
This is all I could find on it. [forum](https://forums.sonarr.tv/t/sonarr-v4-beta/31078)

**Comment by Dulanic** [comment]
LSIO and Hotio have it, could probably see what they do?

https://github.com/linuxserver/docker-sonarr/blob/master/Dockerfile
https://hotio.dev/containers/sonarr/

**Comment by dhenry437** [comment]
Bump. Any further on this? Or could someone point me in the right direction for installing Sonarr 4 using the Debian LXC as a base

**Reply by ghost** [reply]
Upvote. I would be very interested in this too! Getting some errors now when downloading metadata, would be resolved according to the developer in the new version.

**Comment by KickingAss2024** [comment]
Upvote x 2. very interested! The new version is the way as v3 is old news.

**Comment by Dulanic** [comment]
If you guys want this for now, just run docker in a LXC and run LSIO Sonarr v4.

**Comment by tteck** [comment]
To upgrade your Sonarr (v3) LXC to Sonarr v4, you can copy and paste all the commands below at once into the Sonarr LXC console:
```
systemctl stop sonarr.service
wget -q https://download.sonarr.tv/v4/develop/4.0.0.697/Sonarr.develop.4.0.0.697.linux-x64.tar.gz
tar -xzf Sonarr.develop.4.0.0.697.linux-x64.tar.gz
cp -r Sonarr/* /usr/lib/sonarr/bin
rm -rf Sonarr Sonarr.develop.4.0.0.697.linux-x64.tar.gz
sed -i 's|ExecStart=/usr/bin/mono --debug /usr/lib/sonarr/bin/Sonarr.exe -nobrowser -data=/var/lib/sonarr|ExecStart=/usr/lib/sonarr/bin/Sonarr -nobrowser -data=/var/lib/sonarr|' /lib/systemd/system/sonarr.service
systemctl daemon-reload
systemctl start sonarr.service
```

**Reply by towerhand** [reply]
Thank you. You are the bomb.

**Reply by towerhand** [reply]
I just created a new v4 lxc, and for some reason, sonarr is not seeing the NFS folder from my NAS, but I can see and access it from the shell. I have it set up the same way as the other ARRs, and I didn't run into this issue. Any suggestions?

**Reply by tteck** [reply]
Privileged or Unprivileged?

**Reply by Dulanic** [reply]
I havent use NFS in this container, but a mount point works fine here. Just add it to your containers .conf. 
`mp0: /mnt/tank,mp=/mnt/tank`

**Reply by towerhand** [reply]
Privileged and mounted properly as I can see and access the folder from the shell/ssh, but can't map the folder in sonarr because it doesn't see it.

**Reply by Dulanic** [reply]
I see the problem, sonarr in the LXC uses a different user than root. It uses Sonarr. So we can access the files with NFS using root, but /w sonarr you can't. Can we add a user to use to run Sonarr as /w the LXC? I guess the alternative is to ass Sonarr to the host if we need to use privileged anyways for NFS. Or make the full media dir 777, which probably isn't the best idea security wise.

![image](https://github.com/tteck/Proxmox/assets/2948206/2d5da260-8f1b-4d29-9a83-baee12e883dc)

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/discussions/1316#discussioncomment-5584999 ?

**Reply by Dulanic** [reply]
Thank @tteck, Ill work on a workaround, though if we use the v4 installed, dpkg-reconfigure sonarr doesnt work.

**Reply by tteck** [reply]
It initially is installed by apt, just a thought.

**Reply by towerhand** [reply]
Should I create an issue in the sonarr repo?

**Reply by Dulanic** [reply]
It's not a sonarr problem. I switched the service to run by root and its's fine. It's a access issue not a sonarr issue.

`nano /lib/systemd/system/sonarr.service`

and change:
```
[Service]
User=root
Group=root
```

**Reply by tteck** [reply]
I will add the following to the script
```
sed -i 's/\(User=\|Group=\).*/\1root/' /lib/systemd/system/sonarr.service
```

**Reply by Dulanic** [reply]
> I will add the following to the script
> 
> ```
> sed -i 's/\(User=\|Group=\).*/\1root/' /lib/systemd/system/sonarr.service
> ```

As a thought, maybe put that as an option? Y/N flag for would you like to run sonarr as a user other than default? Most probably don't need it changed unless they are using nfs etc.? Might even give them the option to change the user/group?

**Reply by tteck** [reply]
In any circumstance, I can't identify any disadvantages to running as the root user instead of Sonarr.

**Reply by towerhand** [reply]
Thanks guys

**Reply by AlexGeek77** [reply]
For those who might want to upgrade their V4 installation I've updated the commands above. Run these commands in your container to upgrade:
```
systemctl stop sonarr.service
cd /tmp
wget -O SonarrV4.tar.gz 'https://services.sonarr.tv/v1/download/develop/latest?version=4&os=linux'
tar -xzf SonarrV4.tar.gz
cp -r Sonarr/* /usr/lib/sonarr/bin
rm -rf Sonarr SonarrV4.tar.gz
systemctl start sonarr.service
```

**Reply by tteck** [reply]
I'll add that to the script

**Reply by nicoska84** [reply]
I executed those commands but now the Sonarr service is not accessible anymore :(

**Reply by AdmiralAwesome** [reply]
Running update inside LXC with 

`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/sonarr.sh)"`

![obraz](https://github.com/tteck/Proxmox/assets/2470577/5735e0c5-b715-4937-8d01-dd671d29562c)

Will break your Sonarr installation. Website is no longer available.

**Comment by AlexGeek77** [comment]
> In any circumstance, I can't identify any disadvantages to running as the root user instead of Sonarr.

For me, I prefer to have all the *arr apps run from non-root users in the LXCs. I do that so I can more securely bind my CIFS/SMB shares into the unprivileged containers. Sonarr was the only one who used a specific user for the application so I created unique users for all my *arrs. No need to change the script for me, I'll manage it on my side as it's a somewhat unique configuration.

**Reply by Dulanic** [reply]
> > In any circumstance, I can't identify any disadvantages to running as the root user instead of Sonarr.
> 
> For me, I prefer to have all the *arr apps run from non-root users in the LXCs. I do that so I can more securely bind my CIFS/SMB shares into the unprivileged containers. Sonarr was the only one who used a specific user for the application so I created unique users for all my *arrs. No need to change the script for me, I'll manage it on my side as it's a somewhat unique configuration. 
> 
> 

I think a lot do this. However my workaround was forcing the userid with the NFS export.

**Comment by kvmgz** [comment]
Hi folks.
I had v3 Sonarr and thought I give it a shot for the v4.
I tried to update using the initial script. It asks me nicely if I want to upgrade to v4. Everything goes without an error, but I can't access Sonarr anymore.
If I do a clean install and choose v4, it works, but I CBA to reconfigure everything.
Any thoughts?

**Reply by mitchellmarkly** [reply]
I literally just did the same thing, but I believe the script is actually asking "are you updating v4?" as in- you already have v4 installed and are updating. I'm going to try running the commands posted above to upgrade to v4.


*Update: I restored my v3 from a backup and then ran the upgrade commands-worked like a charm.

**Reply by kvmgz** [reply]
You ran the script or what upgrade command are you referring to?

**Comment by towerhand** [comment]
@tteck V4 has been published to the main branch. 
https://www.reddit.com/r/sonarr/s/139hxxhdIh

Is there anything we need to do on the existing lxcs? Or do we need to wait for an update to your script and create a new lxc?

Thanks.

**Reply by AdmiralAwesome** [reply]
> Is there anything we need to do on the existing lxcs? Or do we need to wait for an update to your script and create a new lxc?
![obraz](https://github.com/tteck/Proxmox/assets/2470577/3d03b9cc-23ee-4de8-b1c6-16089bd3fbf7)

> Native versions for Windows, MacOS and Linux have been published and are available to download from [sonarr.tv](https://sonarr.tv/#download). They will need to be downloaded and installed manually, updates will not happen via Sonarr. Note that x86 releases are no longer available for Linux.

**Comment by tteck** [comment]
I have completed a major code refactoring of Sonarr v4 to bring it more inline with the mainstream paths (no more upgrading from v3).
This will be a **Breaking Change**. The previous installs will **not** be updated via the `update` command.
The user is now `sonarr` and the group is `media`.

**Reply by emddummi** [reply]
So New LXC nö Update of my existing v3 lxc?

**Reply by tteck** [reply]
That's correct.

**Reply by tteck** [reply]
I have not tried their installer within the Sonarr LXC console.

```
bash -c "$(wget -qLO - https://github.com/Sonarr/Sonarr/raw/develop/distribution/debian/install.sh)"
```
That may work to bring everything up to date. 🤷🏻‍♂️

**Reply by kvmgz** [reply]
got this message while trying to run those:
groupadd: 'app_guid=$(echo "$app_guid" | tr -d ' ')' is not a valid group name

**Reply by tteck** [reply]
Yeah, they have a code issue. you can download the `install.sh` 
```
wget -q https://raw.githubusercontent.com/Sonarr/Sonarr/develop/distribution/debian/install.sh
```
then execute it locally, `bash install.sh`

**Reply by tteck** [reply]
fixed their installer link, it now works.
```
bash -c "$(wget -qLO - https://github.com/Sonarr/Sonarr/raw/develop/distribution/debian/install.sh)"
```

**Reply by kvmgz** [reply]
Indeed it worked & and all the settings were kept.
Thanks a lot!

Cheers!

**Reply by tteck** [reply]
Now the question is, does the `update` command work? It _should_.

**Reply by kvmgz** [reply]
It did detect sonarr v4, the update worked aswell. And it is working after the process aswell.

**Reply by Jarrekstar** [reply]
So what's the conclusion? Existing instance can be upgraded in place with the official installation shell script?

**Reply by tteck** [reply]
Looks like it. That sure makes things easier.

**Reply by AlexGeek77** [reply]
I was running the develop branch to use v4 already. I updated to use the main branch with this set of commands:
```
systemctl stop sonarr.service
cd /tmp
wget -O SonarrV4.tar.gz 'https://services.sonarr.tv/v1/download/main/latest?version=4&os=linux'
tar -xzf SonarrV4.tar.gz
cp -r Sonarr/* /usr/lib/sonarr/bin
rm -rf Sonarr SonarrV4.tar.gz
systemctl start sonarr.service
```
After that the normal `update` script _should_ work fine.

**Reply by tteck** [reply]
Nope. the `update` command will not work with that setup.

**Reply by AlexGeek77** [reply]
You're correct. I'll build a new LXC at some point in the future to get back to a compliant setup. Due to my user bindings I need to be careful how I set it all up again.

**Reply by tteck** [reply]
You _probably_ can use the Sonarr install script (listed above) to become compliant.

**Reply by kvmgz** [reply]
After successfully upgrading and it was working fine, I realised the downloaded files remain in Que and are not copied to the selected destination folder. If I select the files and try to import manually it says completed but it's not doing anything actually.
Obs.: I used a Turnkey Fileserver to which I associated an SSD drive as a makeshift NAS.
I guess it's some kind of permission issue, since between the Sonarr v3 and 4v some user and group info was changed, but tbh I have no idea what to change or add to solve it :|

**Reply by ianfriend** [reply]
See my comments on this thread https://github.com/tteck/Proxmox/discussions/1738#discussioncomment-8006756

It's probably the same issue of permissions where your download client is creating the files are root:root and now after upgrading to v4 your sonarr is trying to access them as sonarr:media

Fix is to run the install script mentioned in that thread and set sonarr to also run as root:root which is what v3 used to run as

**Reply by kvmgz** [reply]
Yep, tteck's script allowed me to set it back to root:root and all works well.
Thanks for the quick reply mate!

For future ppl lurking here the script was the following:

bash -c "$(wget -qLO - https://github.com/Sonarr/Sonarr/raw/develop/distribution/debian/install.sh)"

**Comment by Jarrekstar** [comment]
Any way to easily copy over the configuration from the existing v3 instance?

**Reply by tteck** [reply]
The data path (`/var/lib/sonarr`) did not change, so you can move the contents.

**Reply by Jarrekstar** [reply]
Awesome, thanks

**Comment by markwoodward86** [comment]
FYI i just took a back up of my old v4 application via the Sonarr UI, ran tteck's latest helper script to start a brand new container, then in the new container i restored from previous backup and voila, i am compliant. Easy as pie.

**Reply by bdoell88** [reply]
I did this and now I'm getting this error:
Cannot install update because startup folder '/opt/Sonarr' is not writable by the user 'sonarr'
![image](https://github.com/tteck/Proxmox/assets/70822874/b0a30270-fcac-4d2c-9378-40d03b2db3cb)
 
Did I do something wrong? All I did was run a backup, run new lxc script, restore from backup.

I also tried updating my existing lxc with the command referenced here https://github.com/tteck/Proxmox/discussions/1738#discussioncomment-8005867

It seemed to update Sonarr, but I am getting the same error as above.

**Reply by markwoodward86** [reply]
Looks like you just need to grant write access to /opt/sonarr for the sonarr user. A quick Google should find the chmod command.

**Reply by bdoell88** [reply]
something weird must be going on. i did a fresh sonarr install and after a while i end up getting the same error. i also can no longer have the webui use port 80. if i try setting it to port 80, the sonarr service will not start and i have to edit the config.xml back to another port for it to run again.

not sure what the problem is. this is very weird. everything else seems to be working fine.

**Reply by maxoss** [reply]
Same issue here. User sonarr doesn't have any access to any folder in the LXC. 

Just did a fresh install `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/sonarr.sh)"`

and did also this https://github.com/tteck/Proxmox/discussions/1738#discussioncomment-8005867

**Reply by tteck** [reply]
Would a `chown -R sonarr:media /opt/Sonarr` fix the issue?

**Reply by maxoss** [reply]
no, it did not

**Reply by tteck** [reply]
As a test do `chmod 777 /opt/Sonarr`, if this works you can roll back permissions to suit your needs.

**Reply by maxoss** [reply]
just picked a few root folder locations:
![sonarr](https://github.com/tteck/Proxmox/assets/757822/02bce75c-b3d0-49a1-ba74-7551d81401e0)

**Reply by nfribeiro** [reply]
I am also getting these permissions errors. Works well in 4.0.0.748.
What can I do?
What was the previous user /group?

**Reply by ianfriend** [reply]
similar permisisons issues here too. i get:

[v4.0.0.748] System.UnauthorizedAccessException: Access to the path '/mediafiles/downloads/completed/{file name redacted for brevity}' is denied.
 ---> System.IO.IOException: Permission denied
 
 linux permissions are not in my skillset - anyone have any idea?  presumably this is down to the user / group change as was working perfectly under v3

UPDATE:
if i do chown -R sonarr:media /mediafiles then the import works, but then fails again for newly downloaded media
using ls -l command it looks like the new folder/file gets added as owner root:root and not sonarr:media

**Reply by ianfriend** [reply]
@maxoss did you find a solution to this in the end?

**Reply by maxoss** [reply]
No, I'm waiting for @tteck because apparently the script doesn't work for new installations

**Reply by tteck** [reply]
What is wrong with new installations?

**Reply by tteck** [reply]
@ianfriend did you upgrade from v3, or a fresh install?

**Reply by ianfriend** [reply]
@tteck i tried both, with the same issue.  the issue appears to be that sabNZBd is putting new files in my download folder as root:root, and now that sonarr is running as sonarr:media it no longer has write access to those new files

**Reply by tteck** [reply]
What's this for? ( I don't use Sonarr)

![Screenshot 2024-01-17 11 55 54 AM](https://github.com/tteck/Proxmox/assets/21342670/c4113942-c8cc-4de5-9afe-9f216e4eeb0c)

**Reply by tteck** [reply]
If you want to control SabNZBd with Sonarr, can you create a dedicated user and configure both Sonarr and SabNZBd to run under that user?

**Reply by ianfriend** [reply]
Interesting, but looks like that might only be for when sonarr owns the file.  I'll give it a try though! (And see if SABNzb has similar settings). Thanks

**Reply by tteck** [reply]
What happens if you execute
```
bash -c "$(wget -qLO - https://github.com/Sonarr/Sonarr/raw/develop/distribution/debian/install.sh)"
```
and change the user and group to root? Just throwing out ideas 🤷🏻‍♂️

**Reply by ianfriend** [reply]
yep thats the fix for me! many thanks!

**Reply by tteck** [reply]
Which one? lol

**Reply by ianfriend** [reply]
Running the install script and setting the user and group to root. No idea why the quote reply never works for me on GitHub

**Reply by tteck** [reply]
I can change the script to not create/use the user sonarr and the group media. If that would be preferred. Just let it use root:root

**Reply by towerhand** [reply]
I agree. I think the other ARRS programs are running as root or used to run as root when I ran the script a little while ago.

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/commit/f6e17e85768302d7d3b27a4c9118b4e63624be33

**Reply by maxoss** [reply]
did a fresh install and it works now, thank

**Reply by bdoell88** [reply]
Is it possible to apply these changes to an existing install instead of doing a fresh install?

**Reply by ianfriend** [reply]
Yes that's what I did.  Use the install script tteck posed above and install as user and group root

**Comment by ghost** [comment]
Anyone else experiencing massive RAM usage for Sonarr v4? Feels like there is a memory leak or something? Currently using 6GB of RAM during scans/searchs...

**Reply by ianfriend** [reply]
not for me, bumbling along the past week at 150-160MB
![image](https://github.com/tteck/Proxmox/assets/8939108/1df8c336-5974-4e83-9505-a2e52d5fee8c)

**Reply by randomparity** [reply]
I've been seeing OOM messages on the host that kill sonarr which is configured for 1GB RAM:

`[Tue Feb 27 12:13:37 2024] oom-kill:constraint=CONSTRAINT_MEMCG,nodemask=(null),cpuset=pvestatd.service,mems_allowed=0,oom_memcg=/lxc/1003,task_memcg=/lxc/1003/ns/init.scope,task=systemd,pid=3459049,uid=100000
[Tue Feb 27 12:13:37 2024] Memory cgroup out of memory: Killed process 3459049 (systemd) total-vm:167644kB, anon-rss:0kB, file-rss:128kB, shmem-rss:0kB, UID:100000 pgtables:84kB oom_score_adj:0
[Tue Feb 27 12:13:39 2024] oom_reaper: reaped process 3459049 (systemd), now anon-rss:0kB, file-rss:128kB, shmem-rss:0kB`

My sonarr version was 4.0.0.740. Sometimes the app is restarted in the container and sometimes I see the container is completely stopped.  I'm trying to update now to see if the issue persists.

**Comment by AkiVonAkira** [comment]
Tried to update Sonarr through the LXC console and the webui, but the first method no longer works and there seems to be a permission issue with the sonarr user.
I would chown but I can't seem to find the startup folder, hehe :p
ty <3
![image](https://github.com/tteck/Proxmox/assets/113895247/2c8d197b-97b1-4c14-8833-3043d6592052)

**Reply by ianfriend** [reply]
https://github.com/Sonarr/Sonarr/raw/develop/distribution/debian/install.sh

Run that script and set user and group back to root. That should solve your problem

**Reply by ghost** [reply]
![image](https://github.com/tteck/Proxmox/assets/89245209/964c5b16-0052-4cf1-8dbf-f7f778208b9a)
Have a similar issue with updating?

**Reply by tteck** [reply]
Did you type `update` in the console?

**Reply by ghost** [reply]
Hi tteck, thanks for your reply. Yes did try the update command, same result

**Reply by tteck** [reply]
This would indicate that you've fallen victim to a breaking change noted here, https://github.com/tteck/Proxmox/commit/a3ef98e1f8cc45f6b42cdebadd0553d79fdd7eaa
To resolve this, you'll need to create a new Sonarr LXC, or continue to use `apt update && apt upgrade` to keep yours up to date.

Seems that you're using V3

**Reply by ghost** [reply]
I am on V4

**Reply by tteck** [reply]
To resolve this, you'll need to create a new Sonarr LXC.
